File: examples/python/logging_example.py
import structlog
import time
from typing import Dict, Any

logger = structlog.get_logger()

class StructuredLogger:
    """Python structured logging example."""
    
    def __init__(self, service: str):
        self.service = service
        self.logger = logger.bind(service=service)
    
    def info(self, msg: str, **kwargs):
        self.logger.info(msg, **kwargs)
    
    def error(self, msg: str, error: Exception, **kwargs):
        self.logger.error(
            msg,
            error_type=type(error).__name__,
            error=str(error),
            **kwargs
        )
    
    def request(self, method: str, path: str, status: int, duration_ms: float):
        self.logger.info(
            "http_request",
            http_method=method,
            http_path=path,
            http_status=status,
            duration_ms=duration_ms
        )

# Example usage
if __name__ == "__main__":
    log = StructuredLogger("example-service")
    log.info("Starting service", version="1.0.0")
    log.request("GET", "/health", 200, 15.2)

File: examples/typescript/logging_example.ts
import pino from 'pino';

interface LoggerConfig {
  service: string;
  level?: string;
}

export class StructuredLogger {
  private logger: pino.Logger;

  constructor(config: LoggerConfig) {
    this.logger = pino({
      name: config.service,
      level: config.level || 'info',
      base: { service: config.service }
    });
  }

  info(msg: string, meta?: Record<string, any>): void {
    this.logger.info(meta, msg);
  }

  error(msg: string, error: Error, meta?: Record<string, any>): void {
    this.logger.error({
      ...meta,
      error_type: error.name,
      error_message: error.message,
      stack: error.stack
    }, msg);
  }

  request(method: string, path: string, status: number, durationMs: number): void {
    this.logger.info({
      http_method: method,
      http_path: path,
      http_status: status,
      duration_ms: durationMs
    }, 'http_request');
  }
}

// Example usage
const log = new StructuredLogger({ service: 'example-service' });
log.info('Starting service', { version: '1.0.0' });
log.request('GET', '/health', 200, 15.2);
